clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py

```diff
@@ -157,6 +170,19 @@ def connect(self):
                         self.clipper_query_port))
                 elif p.name == "7000":
                     self.clipper_rpc_port = p.node_port
+
+            metrics_ports = self._k8s_v1.read_namespaced_service(
+                name="metrics", namespace='default').spec.ports
+            for p in metrics_ports:
+                if p.name == "9090":
+                    self.clipper_metric_port = p.node_port
+                    logger.info("Setting Clipper metric port to {}".format(
+                        self.clipper_metric_port))
+
+            query_addr = "{}:{}".format(self.external_node_hosts[0],
+                                        self.clipper_query_port)
+            start_frontend_exporter(self._k8s_beta, query_addr)
```

Comment:
Why are you starting the frontend exporter in `connect()`? Shouldn't it only be started in `start_clipper`?

Reply:
Fixed.